### PR TITLE
Add required dependencies for ops in scanner on ARM

### LINKED ISSUE #1420: Compiler complains about scanning issues
I was hoping that this will be only on Windows, but when I build on ARM and run publish I still receive following error in output
```
  Generating compatible native code. To optimize for size or speed, visit https://aka.ms/OptimizeCoreRT
  *** Methods compiled but not scanned:
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ULMod(uint64,uint64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.LMod(int64,int64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ULDiv(uint64,uint64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.LDiv(int64,int64)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowLngDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowULngDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowLngArithExc()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.IDiv(int32,int32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.UDiv(uint32,uint32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.IMod(int32,int32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.UMod(uint32,uint32)
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowIntDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowUIntDivByZero()
  [S.P.CoreLib]Internal.Runtime.CompilerHelpers.MathHelpers.ThrowIntArithExc()
  Unhandled exception. System.Exception: Scanning failure
     at ILCompiler.Program.Run(String[] args) in ilc.dll:token 0x600010f+0x1027
     at ILCompiler.Program.Main(String[] args) in ilc.dll:token 0x6000115+0x7
```

### PR DESCRIPTION
**Pull request by kant2002**
ARM using additional helper functions to generate code
Closes #1420

### CODE REVIEW

File: src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs [hunk 1]
```diff
@@ -1069,17 +1069,39 @@ private void ImportBinaryOperation(ILOpcode opcode)
...
+                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.UMod), "_umod");
+                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.Mod), "_mod");
+
+                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.ULDiv), "_uldiv");
+                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.LDiv), "_ldiv");
+                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.UDiv), "_udiv");
+                        _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.Div), "_div");
+                    }
+
                     // Required for ARM64
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs]: Is this one required for ARM32 too?

Would be nice to update the comment if it is (or to just put it into an explicit `if (_compilation...)` block like you have done for ARM32 - we should have done it for the ARM64 specific helpers too.

**Review by MichalStrehovsky — COMMENTED** [review]
Looks good otherwsie, thanks!

File: src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs [hunk 2]
```diff
@@ -1069,17 +1069,39 @@ private void ImportBinaryOperation(ILOpcode opcode)
...
                 case ILOpcode.add_ovf_un:
-                case ILOpcode.mul_ovf:
-                case ILOpcode.mul_ovf_un:
                 case ILOpcode.sub_ovf:
                 case ILOpcode.sub_ovf_un:
                     _dependencies.Add(GetHelperEntrypoint(ReadyToRunHelper.Overflow), "_ovf");
                     break;
+                case ILOpcode.mul_ovf:
+                case ILOpcode.mul_ovf_un:
+                    // Required for ARM
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Compiler/IL/ILImporter.Scanner.cs]:
```suggestion
```
This comment is redundant with the code. The code describes the intent well enough.

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Pull request merged**